spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -25,33 +25,27 @@
 
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
 import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.cloud.gcp.autoconfigure.core.AppEngineCondition;
 import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
 import org.springframework.cloud.gcp.autoconfigure.core.GcpProperties;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Primary;
 import org.springframework.util.StringUtils;
 
 /**
- * Google Cloud SQL starter.
- *
- * <p>Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a database and instance
- * connection name.
+ * Provides Google Cloud SQL instance connectivity through Spring JDBC by providing only a
+ * database and instance connection name.
  *
  * @author João André Martins
  */
 @Configuration
 @EnableConfigurationProperties({ GcpCloudSqlProperties.class, DataSourceProperties.class })
-@ConditionalOnClass(com.google.cloud.sql.mysql.SocketFactory.class)
-@ConditionalOnProperty(name = "spring.cloud.gcp.sql.enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnProperty(
+		name = "spring.cloud.gcp.sql.enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
Don't we want to check that the `spring.datasource.url` has _not_ been set? It feels to me that you should backoff if the user has expressed such an opinion.

Reply:
We don't want to do that here.
The `cloudSqlDataSourceProperties` bean checks for the unlikely case where a user specifies the JDBC URL, then overrides our choice with theirs if they did.

Reply:
That isn't right. Why would you do that?

Reply:
We basically allow users to specify just the instance name and database name, and we build the JDBC URL for them, which is a bit convoluted, or they can specify the full JDBC URL themselves. What's wrong with that?

Reply:
+1 to what @meltsufin said.
Think of it as a simple JDBC URL override, for when a user doesn't like the JDBC URL we provide for whatever reason.

Reply:
I have no problem that you build the URL for them but if the user has provided one already you should not override it.

Reply:
Sorry if I wasn't clear before.
I meant we override the URL we created with the one the user provided, so it seems we're both on the same page!